src/Symfony/Component/Form/Tests/Extension/Core/Type/CheckboxTypeTest.php

```diff
@@ -125,15 +125,15 @@ public function testSubmitWithEmptyValueAndFalseUnchecked()
         $this->assertNull($form->getViewData());
     }
 
-    public function testSubmitWithEmptyValueAndTrueChecked()
+    public function testSubmitWithEmptyValueAndTrueUnchecked()
     {
         $form = $this->factory->create('checkbox', null, array(
             'value' => '',
         ));
         $form->submit(true);
 
-        $this->assertTrue($form->getData());
-        $this->assertSame('', $form->getViewData());
+        $this->assertNull($form->getData());
```

Comment:
This is a BC break for people who handle APIs with the Form component. I think you should handle `true` and `false` as well (and reverse-transform them to `true` and `false`).